exercises/05.actions/05.solution.history-revalidation/server/app.js

```diff
@@ -13,54 +11,73 @@ import { shipDataStorage } from './async-storage.js'
 
 const PORT = process.env.PORT || 3000
 
-const app = express()
+import { Hono } from 'hono'
+import { compress } from 'hono/compress'
+import { serveStatic } from '@hono/node-server/serve-static'
+import { serve } from '@hono/node-server'
+
+const app = new Hono()
+
 app.use(compress())
 // this is here so the workshop app knows when the server has started
-app.head('/', (req, res) => res.status(200).end())
+app.get('/', () => new Response())
 
-app.use(express.static('public', { index: false }))
-app.use('/js/src', express.static('src'))
+app.use(serveStatic({ root: 'public', index: false }))
+app.use('/js/src', serveStatic({ root: 'src' }))
 
 // This just cleans up the URL if the search ever gets cleared... Not important
 // for RSCs... Just ... I just can't help myself. I like URLs clean.
-app.use((req, res, next) => {
-	if (req.query.search === '') {
-		const searchParams = new URLSearchParams(req.search)
+app.use(({ req, redirect }) => {
+	if (req.query('search') === '') {
+		/**
+		 * @fixme Not sure what Kent means by
+		 * `req.query.search` and `req.search`.
+		 */
+		const searchParams = new URLSearchParams()
```

Comment:
This is the only place I didn't figure out. There's some difference between `req.search` and `req.query.search` in Express, and I don't know it what that is.

Reply:
I can't find `req.search` in [Express API docs](https://expressjs.com/en/api.html). Hmm.

Reply:
@kentcdodds, was this logic perhaps originally flawed a bit? Shouldn't it be this:

```js
if (req.query.search !== '') {
  // Then remove "search" from query
  // and redirect.
}
```